react/react_intro/README.md

```diff
@@ -28,41 +28,53 @@
 
 
-There are at least four entirely distant components that are on the page including:
+There are at least four entirely distant UI elements on the page, including:
 
 - the top bar
 - the left sidebar
-- the stories feed
-- the contacts list.  
+- the stories and feed section
+- the contacts list
 
-Using our DOM manipulation approach, we would have all of these different elements in a single html file.  This would be challenging to maintain because we a bug in one area could easily affect the rest of the page.  
+These are illustrated here:
 
-React solves this problem by introducing `Components`.  Instead of laying out all of our HTML in a single file, we can separate it into separate classes and combine those classes together.
+![Facebook redesign separated into components](./images/fbInterface.jpg)
+
+Using our DOM manipulation approach, we would have all of these different elements in a single HTML file. This would be challenging to maintain because a bug in one area could easily affect the rest of the page. Making matters worse, the bug would be difficult to find. A developer would have to sift through a single huge file to find the issue - think finding a needle in a haystack.
+
+React solves this problem by introducing `Components`. Instead of laying out all of our HTML and JS in single files, we can separate them into separate files based on what part of the UI they are responsible for rendering. We then import these files, organize them, and let React combine their markup and functionality into a single, cohesive page. If we have an issue in one component, we simply open that component's file, and the code that we need to fix is right there.
+
+Remember, however, that a web browser can't read and combine components by itself. Even if a developer would like neatly separated files, the browser still needs a single HTML page. The aim of React is to improve quality-of-life for developers while packaging frontend applications in a format that browsers can still read and render efficiently.
 
 ## Virtual DOM
 
-How does React build a single HTML page from multiple different components?  A first approach might be that whenever a component adds an HTML element (e.g creating a new comment), it redraws the entire DOM.  While this would work, the process would be quite slow.  The solution that React uses is to create a **virtual DOM** that represents the DOM.  Whenever one component changes, it updates that virtual DOM.  The VDOM then sees what changes were made, then only updates the DOM with those changes.  This way, the whole DOM doesn't need to be recreated.
+So, React assembles an HTML file from all of these different components. How, then, does it change that file when something on the page changes? Say I add a comment on somebody's Facebook post. How does everything update?
```

Comment:
Is it 100% accurate to say that React assembles an HTML file?

Reply:
It's funny, my production app does because it uses React for SSR. However, you're right - on the client, React assembles a DOM tree, which is rendered via HTML to the user. I can add this. The more under the hood stuff we add, however, the more clarity we potentially lose.